packages/puppeteer-core/src/cdp/BrowserContext.ts

```diff
@@ -100,6 +102,33 @@ export class CdpBrowserContext extends BrowserContext {
     await this.#browser._disposeContext(this.#id);
   }
 
+  override async cookies(): Promise<Cookie[]> {
+    const {cookies} = await this.#connection.send('Storage.getCookies', {
+      browserContextId: this.#id || undefined,
+    });
+    return cookies.map(cookie => {
+      return {
+        ...cookie,
+        partitionKey: cookie.partitionKey
```

Comment:
I believe we want to return the complete partitionKey here and not only the topLevelSite.

Reply:
You are correct, fixed by returning the cookies directly (without this modification)